File: benchmarks/solutions/swift/r5-http-server.swift
import Foundation
import Vapor

let app = try Application(.detect())

app.get("hello") { req -> Response in
    let name = req.query[String.self, at: "name"] ?? "World"
    let body = try JSONEncoder().encode(["greeting": "Hello, \(name)!"])
    return Response(
        status: .ok,
        headers: ["Content-Type": "application/json"],
        body: .init(data: body)
    )
}

app.middleware = []
app.any("*") { _ -> Response in
    let body = try JSONEncoder().encode(["error": "not found"])
    return Response(
        status: .notFound,
        headers: ["Content-Type": "application/json"],
        body: .init(data: body)
    )
}

app.http.server.configuration.port = 8080
try app.run()

File: benchmarks/solutions/ruby/r5-http-server.rb
require "webrick"
require "json"

server = WEBrick::HTTPServer.new(Port: 8080, Logger: WEBrick::Log.new("/dev/null"))

server.mount_proc "/hello" do |req, res|
  name = req.query["name"] || "World"
  res["Content-Type"] = "application/json"
  res.body = JSON.generate(greeting: "Hello, #{name}!")
end

server.mount_proc "/" do |req, res|
  res.status = 404
  res["Content-Type"] = "application/json"
  res.body = JSON.generate(error: "not found")
end

server.start
